pkg/backend/httpstate/client/ai.go

```diff
@@ -54,8 +66,37 @@ func createSummarizeUpdateRequest(
 	}
 }
 
-// extractSummaryFromResponse parses the Copilot API response and extracts the summary content
-func extractSummaryFromResponse(copilotResp apitype.CopilotSummarizeUpdateResponse) (string, error) {
+// createExplainPreviewRequest creates a new CopilotExplainPreviewRequest with the given content and org ID
+func createExplainPreviewRequest(
+	lines []string,
+	orgID string,
+	maxUpdateOutputLen int,
+) apitype.CopilotExplainPreviewRequest {
+	content := strings.Join(lines, "\n")
+	content = TruncateWithMiddleOut(content, maxUpdateOutputLen)
+
+	return apitype.CopilotExplainPreviewRequest{
+		CopilotRequest: apitype.CopilotRequest{
+			State: apitype.CopilotState{
+				Client: apitype.CopilotClientState{
+					CloudContext: apitype.CopilotCloudContext{
+						OrgID: orgID,
+						URL:   "https://app.pulumi.com",
```

Comment:
Do we have to set this URL?

Reply:
Yes, though it doesn't make so much sense. Will look at relaxing that for api calls.